src/Microsoft.AspNetCore.Razor.Evolution/RazorSourceDocument.cs

```diff
@@ -9,6 +9,16 @@ namespace Microsoft.AspNetCore.Razor.Evolution
 {
     public abstract class RazorSourceDocument
     {
+        public abstract Encoding Encoding { get; }
```

Comment:
I feel like it'd be too early to do docs until we know the final form of the system. Also, in the past we've avoided documenting Razor bits because of their limited use.

When things are more complete I don't think it'd be a bad idea to break the latter rule since the whole purpose of Razor.Evolution is to make Razor more consumable by users.